tests/test_endgame_tablebase.py

```diff
@@ -15,21 +15,42 @@ def move_from_et_query(fen: str):
 
 
 class TestEndgameTablebase:
-    def _check_et_query_move_expected(self, fen: str, move_expected: bool):
+    def _check_et_query_move_expected(
+        self, test_name: str, fen: str, move_expected: bool
+    ):
         move = move_from_et_query(fen)
         if move_expected:
-            assert move, "Expected move but none returned."
+            assert move, f"{test_name}: Expected move but none returned."
         else:
-            assert not move, "Didn't expect move but returned valid move."
+            assert not move, f"{test_name}: Didn't expect move but returned valid move."
 
-    def test_et_query_white_to_move_white_winning(self):
-        self._check_et_query_move_expected("8/4k3/8/8/8/8/3BB3/3K4 w - - 0 1", True)
+    @pytest.mark.parametrize(
+        "test_name, fen, move_expected",
+        [
+            ("white_to_move_white_winning", "8/4k3/8/8/8/8/3BB3/3K4 w - - 0 1", True),
+            ("black_to_move_white_winning", "8/4k3/8/8/8/8/3BB3/3K4 b - - 0 1", False),
+            (
+                "black_to_move_white_winning_extra_queen",
+                "8/4k3/8/8/8/8/3BB3/3K1Q2 b - - 0 1",
+                False,
+            ),
+        ],
+    )
+    def test_et_query(self, test_name: str, fen: str, move_expected: bool):
+        self._check_et_query_move_expected(test_name, fen, move_expected)
 
-    def test_et_query_black_to_move_white_winning(self):
-        self._check_et_query_move_expected("8/4k3/8/8/8/8/3BB3/3K4 b - - 0 1", False)
-
-    def test_et_query_black_to_move_white_winning_extra_queen(self):
-        self._check_et_query_move_expected("8/4k3/8/8/8/8/3BB3/3K1Q2 b - - 0 1", False)
+    def test_2nd_probe(self):
+        board = chess.Board()
+        board.set_fen("8/4k3/8/8/8/8/3BB3/3K4 w - - 0 1")
+        et = EndgameTablebase(EndgameTablebaseConfig("data/endgame_tablebases"))
+        move = et.query(board)
+        board.push(move)
+        # Play black move
+        board.push(next(iter(board.legal_moves)))
+        # Check there is a move returned on second probe
+        # Cannot check dtz(move) > dtz(move2), this isn't public API unfortunately
+        move2 = et.query(board)
+        assert move2
 
```

Comment:
I am in 2 minds...we have discussed before that it is not ideal to test private (or in our case, protected) member functions - see [here](https://stackoverflow.com/questions/105007/should-i-test-private-methods-or-only-public-ones). But the majority of our code is private (protected) methods - if we never test them, there is no way we would achieve anywhere near 70% coverage.

I feel a bit uncomfortable if our unit test coverage is, say below 20%, we would struggle to understand the behaviour of our code and also make code refactoring a nightmare.

Thoughts? @michelle-lai-swl @KYLChiu

Reply:
As I understood your public functions should call your private functions - so these should get covered anyway?